### PR TITLE
Fix docs build error and warnings

### PR DESCRIPTION
**Pull request by neon60**
@parbenc please continue, fix all warnings.

### CODE REVIEW

**Comment by neon60** [comment]
@samjwu There are some overlap between your spelling branch and our branch. Can you check our changes and suggest if we should take over your commit(s)? We can also add the changes to your branch, if you want.

**Comment by samjwu** [comment]
> @samjwu There are some overlap between your spelling branch and our branch. Can you check our changes and suggest if we should take over your commit(s)? We can also add the changes to your branch, if you want.

I think it might be clearest to have both PRs to focus on separate concerns. It looks like this one fixes some things with doxygen while mine only does spelling.

**Review by samjwu — DISMISSED** [review]

**Comment by neon60** [comment]
> > @samjwu There are some overlap between your spelling branch and our branch. Can you check our changes and suggest if we should take over your commit(s)? We can also add the changes to your branch, if you want.
> 
> I think it might be clearest to have both PRs to focus on separate concerns. It looks like this one fixes some things with doxygen while mine only does spelling.

You are right. We will have separate PRs and your PR already merged in.

**Comment by parbenc** [comment]
Separated spellcheck stuff into another branch.

**Comment by neon60** [comment]
> Separated spellcheck stuff into another branch.

@parbenc Can you open that PR?

**Comment by neon60** [comment]
@parbenc Can we remove the WIP from the title?

File: library/include/hiptensor/internal/type_traits.hpp [hunk 1]
```diff
@@ -270,6 +271,68 @@ namespace std
         hiptensor::detail::Fp16Bits eps(static_cast<uint16_t>(0x7FC0));
         return eps.b16;
     }
+
+    ///////////////////////////////////////////////////////////
```
Comment (neon60) [inline library/include/hiptensor/internal/type_traits.hpp]: Seems to be new functions added? Do we need these?

Comment (parbenc) [reply library/include/hiptensor/internal/type_traits.hpp]: you're right, this has sneaked back with the rebase. i have removed it

**Comment by parbenc** [comment]
> @parbenc Can we remove the WIP from the title?

Unfortunately i can't edit this PR.

**Comment by neon60** [comment]
Looks good to me.

**Comment by parbenc** [comment]
@cgmillette could you please look at this?

File: library/include/hiptensor/internal/type_traits.hpp [hunk 2]
```diff
@@ -270,6 +271,8 @@ namespace std
         hiptensor::detail::Fp16Bits eps(static_cast<uint16_t>(0x7FC0));
         return eps.b16;
     }
+#endif // DOXYGEN_SHOULD_SKIP_THIS
+    // @endcond
```
Comment (cgmillette) [inline library/include/hiptensor/internal/type_traits.hpp]: Need the endcond?

**Review by cgmillette — COMMENTED** [review]
LGTM - just not sure about the dangling endcond

File: library/include/hiptensor/internal/type_traits.hpp (hunk 2 above)
Comment (parbenc) [reply library/include/hiptensor/internal/type_traits.hpp]: probably not. it snuck back in during a rebase

Comment (parbenc) [reply library/include/hiptensor/internal/type_traits.hpp]: removed it. could you please approve it again

**Review by cgmillette — APPROVED** [review]

**Review by saadrahim — APPROVED** [review]

**Pull request merged**